proto/internal/temporal/server/api/taskqueue/v1/message.proto

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020 Temporal Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.server.api.taskqueue.v1;
+
+option go_package = "go.temporal.io/server/api/taskqueue/v1;taskqueue";
+
+import "google/protobuf/empty.proto";
+
+// TaskVersionDirective controls how matching should direct a task.
+message TaskVersionDirective {
+    // Default value (if directive is not present):
+    // "unversioned": use the unversioned task queue, even if the task queue has
+    // versioning data.
+    oneof directive {
```

Comment:
Personally, I don't love this type of stuttering, always looks weird to me to have the same noun repeated.
I'd rename this to `variant`, `directive.variant` reads better than `directive.directive`.

Reply:
I dunno, it's stuttering vs introducing a whole new word that might make readers think is a different concept...

I looked through our existing protos and I see `value` being used for this purpose, which reads better to me